docs: add documentation for `redact_query_validation_errors` config option (#8891)
apollographql/router: docs/source/routing/configuration/yaml.mdx

  introspection_max_depth: false
```

#### Query validation error redaction

> Query validation errors might reveal information about your schema structure, such as field names, types, and required arguments. Although this is not typically a significant security concern (especially with persisted queries), you might want to redact these error details in production environments.

Reviewer: ```suggestion
Query validation errors might reveal information about your schema structure, such as field names, types, and required arguments. Although this is not typically a significant security concern (especially with persisted queries), you can redact these error details in production environments.
```

> Query validation errors can potentially reveal information about your schema structure, such as field names, types, and required arguments. While this is not typically a significant security concern (especially with persisted queries), you may want to redact these error details in production environments.

Reviewer: ```suggestion
Query validation errors might reveal information about your schema structure, such as field names, types, and required arguments. Although this is not typically a significant security concern (especially with persisted queries), you might want to redact these error details in production environments.
```
